back/spec/acceptance/request_codes_spec.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rspec_api_documentation/dsl'
+
+resource 'Request codes' do
+  before do
+    set_api_content_type
+    SettingsService.new.activate_feature! 'user_confirmation'
+  end
+
+  post 'web_api/v1/user/request_code_unauthenticated' do
+    with_options scope: :request_code do
+      parameter :email, 'The email of the user requesting a confirmation code.', required: true
+    end
+
+    before do
+      allow(RequestConfirmationCodeJob).to receive(:perform_now)
+    end
+
+    example 'It works with a passwordless user' do
+      user = create(:user_no_password)
+      do_request(request_code: { email: user.email })
+      expect(RequestConfirmationCodeJob).to have_received(:perform_now).with(user).once
+    end
+
+    example 'It does not work with a user with password' do
+      user = create(:user)
+      do_request(request_code: { email: user.email })
+      expect(RequestConfirmationCodeJob).not_to have_received(:perform_now)
+    end
+
+    example 'It does not work if user reached email_confirmation_code_reset_count' do
+      user = create(:user_no_password, email_confirmation_code_reset_count: 4)
+      do_request(request_code: { email: user.email })
+      expect(RequestConfirmationCodeJob).not_to have_received(:perform_now)
+    end
+
+    example 'Clears the new_email if present' do
```

Comment:
This might be outside the scope of the PR, but I think ideally we should use the `new_email` field for new users as too.